### PR TITLE
Implement expit

### LINKED ISSUE #27: RFC: add the inverse logit function
## Checklist

> Please ensure the following tasks are completed before filing an issue.

* [x] Read and understood the [Code of Conduct][code-of-conduct].
* [x] Searched for existing issues and pull requests.
* [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
* [x] If this is a feature request, the issue name begins with `RFC: `.

## Description

> Description of the issue (or feature request).

This RFC proposes adding support for computing the [inverse logit function](https://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.special.expit.html). The implementation is straightforward:

```javascript
/**
* @param {number} x - input value
* @returns {number} function result
*/
function expit( x ) {
    return 1.0 / ( 1.0 + exp( -x ) );
}
```

Package: `@stdlib/math/base/special/expit`
Alias: `base.expit`

## Related Issues

> Does this issue (or feature request) have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this issue (or feature request)? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

Reference implementations:

-   SciPy \[[1](https://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.special.expit.html)]\[[2](https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/special/_logit.c.src#L31)]

See also `@stdlib/math/base/special/logit`.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md
[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by MatthewCochrane**
Resolves #27  .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:
 
- Implements the expit function.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #27 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Ended up using `1.5 * EPS` for the tests as discussed in the gitter chat.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/benchmark.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+/**
+* Evaluates the expit function.
+*
+* @param x   input value
+* @return result
+*/
+double expit( double x ) {
+	return 1.0 / ( 1.0 - exp( -x ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/benchmark.c]: 1-exp(-x) or 1+exp(-x)?

File: lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+{{alias}}( x )
+    Evaluates the expit function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt]:
```suggestion
    Evaluates the standard logistic function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    Otherwise know as the standard logistic function
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+{{alias}}( x )
+    Evaluates the expit function.
+
+    Otherwise know as the standard logistic function 
+    or sigmoid function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Evaluates the expit function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts]:
```suggestion
* Evaluates the standard logistic function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+*
+* Also called the standard logistic function or the sigmoid function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+*
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts:23]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+* @example
+* var y = expit( Infinity );
+* // returns 1.0
+*
+* @example
+* var y = expit( NaN );
+* // returns NaN
+*/
+declare function expit( p: number ): number;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/docs/types/index.d.ts]:
```suggestion
declare function expit( x: number ): number;
```

File: lib/node_modules/@stdlib/math/base/special/expit/lib/expit.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var exp = require( '@stdlib/math/base/special/exp' );
+
+
+// MAIN //
+
+/**
+* Evaluates the expit function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/lib/expit.js]:
```suggestion
* Evaluates the standard logistic function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/lib/index.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Evaluate the expit function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/lib/index.js]:
```suggestion
* Evaluate the standard logistic function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/package.json [hunk 11]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@stdlib/math/base/special/expit",
+  "version": "0.0.0",
+  "description": "Expit function.",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/package.json]:
```suggestion
  "description": "Standard logistic function.",
```

File: lib/node_modules/@stdlib/math/base/special/expit/test/test.js [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+	t.end();
+});
+
+tape( 'the function returns `0.0` when provided `-Infinity`', function test( t ) {
+	var y = expit( NINF );
+	t.equal( y, 0.0, 'returns 0.0' );
+	t.end();
+});
+
+tape( 'the function evaluates the expit of `x` for negative numbers', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/test/test.js]:
```suggestion
tape( 'the function evaluates the standard logistic function for negative numbers', function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/expit/test/test.js [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.5 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. v: '+y+'. E: '+expected[i]+' Δ: '+delta+'. tol: '+tol );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the expit of `x` for positive numbers', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/test/test.js]:
```suggestion
tape( 'the function evaluates the standard logistic function for positive numbers', function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/expit/README.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+## Usage
+
+```javascript
+var expit = require( '@stdlib/math/base/special/expit' );
+```
+
+#### expit( x )
+
+Computes the [expit][standard-logistic] function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/README.md]:
```suggestion
Computes the [standard logistic][logistic-function] function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/README.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+-->
+
+# expit
+
+> Compute the [standard logistic][standard-logistic] function.
+
+<section class="intro">
+
+The [Standard Logistic][standard-logistic] function, also called the expit function or sigmoid function, is defined as the [logistic][standard-logistic] function with parameters (k = 1, x0 = 0, L = 1).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/README.md]:
```suggestion
The [standard Logistic][logistic-function] function, also called the expit function, inverse logit, or sigmoid function, is defined as the [logistic][logistic-function] function with parameters (`k = 1`, `x0 = 0`, `L = 1`).
```

File: lib/node_modules/@stdlib/math/base/special/expit/README.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# expit
+
+> Compute the [standard logistic][standard-logistic] function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/README.md]:
```suggestion
> Compute the [standard logistic][logistic-function] function.
```

File: lib/node_modules/@stdlib/math/base/special/expit/README.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+
+</section>
+
+<!-- /.examples -->
+
+<section class="links">
+
+[standard-logistic]: https://en.wikipedia.org/wiki/Logistic_function
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/README.md]:
```suggestion
[logistic-function]: https://en.wikipedia.org/wiki/Logistic_function
```

File: lib/node_modules/@stdlib/math/base/special/expit/test/fixtures/python/runner.py [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+    filepath = os.path.join(DIR, name)
+
+    # Write the data to the output filepath as JSON:
+    with open(filepath, "w") as outfile:
+        json.dump(data, outfile)
+
+
+def main():
+    """Generate fixture data."""
+    x = np.logspace(np.log10(0.0001), np.log10(1e5), 5000)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/test/fixtures/python/runner.py]:
```suggestion
    x = np.logspace(np.log10(0.0001), np.log10(1.0e5), 5000)
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@MatthewCochrane Wow! This PR looks great. Just some minor documentation changes and a possible benchmark bug. Once resolved, this should be ready for merge! Thanks for working on this!

File: lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/benchmark.c (hunk 1 above)
Comment (MatthewCochrane) [reply lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/benchmark.c]: Good catch.  It should be 1/(1-exp(x)).  I've updated it in another commit.

**Review by MatthewCochrane — COMMENTED** [review]
Thanks for the feedback.  I've made the changes you requested and fixed that benchmarking bug.  I haven't actually run the c code.  I suppose i need to `make install deps` to get boost to be able to compile it?  What's the purpose of the benchmarks in other languages by the way?  I take it to compare with our javascript implementation?  But there is no automated way to run the other language benchmarks yet?

**Comment by kgryte** [comment]
@MatthewCochrane To run C benchmarks,

```
$ make benchmark-c BENCHMARKS_FILTER=.*/expit/.*
```

In order to run these C benchmarks, you do not need to install Boost. You only need Boost to run benchmarks against Boost, which the C benchmarks in this PR does not use.

Note: in order to compile C benchmarks, you will need `gcc` installed, which, if you are not on Windows, should not be required.

And you are correct. The benchmarks in the other languages are included in order to compare implementations to (1) reference implementations and (2) baseline performance implementations. In this case, including a C benchmark is useful for establishing baseline performance and assessing how much a JavaScript implementation deviates from the baseline.

**Comment by MatthewCochrane** [comment]
Ahh great. Good to know, thanks.  Yep, makes sense.

**Comment by kgryte** [comment]
@MatthewCochrane No problem! Taking a final look at the PR now!

**Review by kgryte — APPROVED** [review]

**Pull request merged**